src/openrct2-ui/windows/Scenery.cpp

```diff
@@ -997,6 +1043,77 @@ class SceneryWindow final : public Window
         }
     }
 
+    void SetFilteredScenery(const size_t tabIndex)
+    {
+        auto currentTab = _tabEntries[tabIndex];
+
+        _filteredScenery.clear();
+        for (auto selection : currentTab.Entries)
+        {
+            if (IsFiltered(selection))
+                _filteredScenery.push_back(selection);
+        }
+    }
+
+    bool IsFiltered(const ScenerySelection& selection)
+    {
+        auto empty = _filter.empty();
+        auto length = _filter.length();
+        if (empty || length == 0)
+            return true;
+
+        const SceneryEntryBase* sceneryEntry = nullptr;
```

Comment:
You should get the object not the entry. Use `Context::GetObjectManager::GetLoadedObject`

Reply:
done